File: luhn.py
def luhn(card_number):
    card_number = [int(x) for x in card_number.split()]
    result = sum(
        (i % 2 == 0 and int(digit) * 2) or int(digit)
        for i, digit in enumerate(card_number)
    )
    return str(result)

File: luhn.cpp
#include <vector>
 
int luhn(const std::vector<int>& input) {
    int sum = 0;
    bool alternate = false;

    for (int i = input.size() - 1; i >= 0; --i) {
        int digit = input[i];
        if (alternate) {
            digit *= 2;
            if (digit > 9) digit -= 9;
        }
        sum += digit;
        alternate = !alternate;
    }

    return sum;
}